### PR TITLE
Feature: Flat Design

### PR DESCRIPTION
**Pull request by Fischaela**
New features:

* Button Formats: Rectangle (default), Square, Cover
* Button Styles: Filled (default), Outline, Frameless
* Button Color: Dynamic colors out of one color attribute
* Flat Button and Popup Styles
* Popup has spatial interface
* Add autoprefixer

### CODE REVIEW

File: src/coffee/button.coffee [hunk 1]
```diff
@@ -6,29 +6,57 @@ Colors = require('./colors.coffee')
...
+    if @options.format != 'square'
+      @buttonHtml = "<span>#{@I18n.t('button')}</span>"
+    else if @options.format == 'square'
+      @elem.addClass('square')
+
+    # Check if data-format is cover
+    # and provide a logo element variable for the image
+    if @options.format == 'cover'
+      @logoElem = $('#podlove-subscribe-button-logo')
+
```
Comment (benzimmer) [inline src/coffee/button.coffee]: Maybe move the two `if`s into a function `addFormat`.

File: src/coffee/app.coffee [hunk 2]
```diff
@@ -39,14 +39,32 @@ class SubscribeButton
...
-      colors: new Colors(@scriptElem.data('colors'))
       buttonId: @scriptElem.data('buttonid')
       hide: @scriptElem.data('hide')
+      style: @scriptElem.data('style')
+      format: @scriptElem.data('format')
+
+    # fallback for old color settings
+    # if new color setting "data-color" is available, use it,
+    # if not, use the old "data-colors"
+    if @scriptElem.data('color')
```
Comment (benzimmer) [inline src/coffee/app.coffee]: Nice!

File: src/coffee/clients_panel.coffee [hunk 3]
```diff
@@ -148,14 +143,8 @@ class ClientsPanel extends Panel
...
     <div>
-      <div class="top-bar">
-        <span class="podlove-subscribe-back-button">&lsaquo;</span>
-        <img src="{{scriptPath}}/images/icon-big@2x.png">
-        <span class="panel-title">{{t "panels.title"}}</span>
-      </div>
       <div class="device-cloud-switch">
-        <button class="podlove-subscribe-local active">{{t "clients_panel.app"}}</button>
-        <button class="podlove-subscribe-cloud">{{t "clients_panel.cloud"}}</button>
+        <button class="podlove-subscribe-local active">{{t "clients_panel.app"}}<span class="podlove-subscribe-tab-active"></span></button><button class="podlove-subscribe-cloud">{{t "clients_panel.cloud"}}<span class="podlove-subscribe-tab-active"></span></button>
```
Comment (benzimmer) [inline src/coffee/clients_panel.coffee]: Some linebreaks between elements please :)

File: src/coffee/popup.coffee [hunk 4]
```diff
@@ -69,13 +123,14 @@ class Popup
...
-    @clientsPanel.moveLeft(amount)
-
-  movePodcast: (amount) ->
-    @podcastPanel.moveLeft(amount)
-
-  moveFinish: (amount) ->
-    @finishPanel.moveLeft(amount)
+  movePanels: (step) ->
+    @container = @elem.find('#podlove-subscribe-popup-modal-inner')
+    if @container.hasClass('swiped-left-0')
```
Comment (benzimmer) [inline src/coffee/popup.coffee]: I think the `if`s can be removed here. `removClass` won't complain if the class is not there.

**Comment by benzimmer** [comment]
Really great work! Only a few minor things in the code.

Any chance we can get rid of the scroll-bar in the help panel? Less padding and smaller font size perhaps?

How much effort would it be to also use the button colors for the highlights in the popup?

**Comment by Fischaela** [comment]
> How much effort would it be to also use the button colors for the highlights in the popup?
Which highlights do you mean?

File: src/coffee/button.coffee (hunk 1 above)
Comment (Fischaela) [inline src/coffee/button.coffee]: Good, idea, that's better!

File: src/coffee/clients_panel.coffee (hunk 3 above)
Comment (Fischaela) [inline src/coffee/clients_panel.coffee]: Okay, sure. I had to add comments instead, to make the two `display: inline-block` and `width: 50%` buttons fit into one line.

File: src/coffee/popup.coffee (hunk 4 above)
Comment (Fischaela) [inline src/coffee/popup.coffee]: Awesome, you're right!

**Comment by Fischaela** [comment]
Too bad. Even if I reduce the `line-height` and `margins` of paragraphs and headline and the top and bottom `paddings` to a minimum, the english version will be still scrollable. :/ I do not want to reduce the left and right `paddings`, because I wanted the text to be centered and scrollable next to the fixed close button. So, because there are more languages, I would prefer to as it like it is.

popup_help

**Comment by Fischaela** [comment]
Ah, ich weiß glaube ich, was du meinst. Macht keinen Aufwand. Hab ich eingefügt.

bildschirmfoto 2016-03-26 um 10 39 42

**Comment by benzimmer** [comment]
This is also nice, but I meant for example the color of the buttons and hover states in the popup:

screen1
screen2

**Comment by timpritlove** [comment]
Can you enlarge the text field for the URL (for copying) to span multiple lines? URLs can be quite long and copying text in horizontally scrolling fields it's particularly complicated. It's better to show as much as possible

Adding a pure HTML link to the feed at that stage might also be an option given the fact that copying real links is usually easier. Not sure if this would make the whole thing too complicated.

**Comment by Fischaela** [comment]
@benzimmer I updated the pr with a more compact layout for the help panel.

@timpritlove I don't know, if you saw the gif. Clicking in the input field marks the whole text, so a user only has to click and copy it (with `cmd + c`). I thought this was an established and easy pattern. However, if you prefer a multi-line text field with line broken url text, I can implement it, of course.

podlove_copy-paste

**Pull request merged**